compat/mingw.c

```diff
@@ -3361,9 +3401,22 @@ static void setup_windows_environment(void)
 		convert_slashes(tmp);
 	}
 
-	/* simulate TERM to enable auto-color (see color.c) */
-	if (!getenv("TERM"))
-		setenv("TERM", "cygwin", 1);
+
+	/*
+	 * Make sure TERM is set up correctly to enable auto-color
+	 * (see color.c .) Use "cygwin" for older OS releases which
+	 * works correctly with MSYS2 utilities on older consoles.
+	 */
+	if (!getenv("TERM")) {
+		if (windows_build() >= 15063) {
+			setenv("TERM", "xterm-256color", 1);
+
+			if (!getenv("COLORTERM"))
+				setenv("COLORTERM", "truecolor", 1);
+		}
+		else
+			setenv("TERM", "cygwin", 1);
```

Comment:
It would save a couple brain cycles to have the shorter conditional arm first, and also to use `0` as the `override` parameter of `setenv()`, i.e.:

```c
if (!getenv("TERM")) {
  if (GetVersion() < 15063 << 16)
    setenv("TERM", "cygwin", 0);
  else {
    setenv("TERM", "xterm-256color", 0);
    setenv("COLORTERM", "truecolor", 0);
  }
}
```